### ISSUE TITLE
radcan possible bug with trig functions

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2882]
Imported from SourceForge on 2024-07-07 00:14:56
Created by **[j-p-o](https://sourceforge.net/u/j-p-o/)** on 2021-09-24 02:38:34
Original: https://sourceforge.net/p/maxima/bugs/3862

---

I only use maxima a couple of times per year, so I may have a pilot error. However, it seems that "radcan" may have a bug. I am using it to compare the differentiated result of an integral against the expression that is integrated.

After using "ratsimp" on the differentiated integral result  - which when evaluated for a specific set of parameters shows equality with the expression to be integrated - "radcan" is invoked and equality is lost. 

Thank you for any help - hope I have submitted this in the proper manner.

Attachments:
- https://sourceforge.net/p/maxima/bugs/3862/attachment/Integration_Differentiation_radcan.wxmx

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 00:14:57
Created by **[richardgobeli](https://sourceforge.net/u/richardgobeli/)** on 2021-12-09 05:13:21
Original: https://sourceforge.net/p/maxima/bugs/3862/#218f

---

I have seen a similar result in other cases using ctensor and sqrt factors of differences.
When radcan switches a sign of a factor in the sqrt to cancel with another factor, it should multiply the sqrt by *i* so that this sign change is remembered.  If an even number of sign changes of divison were made the *i* s would cancel. but if an even number of sign changes of products made a square the result should be the product of two *i* , which will be minus and not a plus.  If it chooses plus instead of minus the result has changed.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 00:15:01
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2021-12-09 15:02:12
Original: https://sourceforge.net/p/maxima/bugs/3862/#9c59

---

This is correct behavior by radcan. The bug is that it is not documented in the manual.
See [the original paper](https://people.eecs.berkeley.edu/~fateman/papers/radcan.pdf) or [this discussion thread](https://def.fe.up.pt/pipermail/maxima-discuss/2010/034274.html).

Basically, the problem is that ``sqrt`` is a multivalued function.

A simpler example where radcan(ex) doesn't match ex numerically:
~~~~
ex: sqrt(x^2-1)/sqrt(1-x)$
exr: radcan(ex)$
[ex,exr],x=0 => [%i,-%i]
~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 00:15:04
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2021-12-09 15:02:46
Original: https://sourceforge.net/p/maxima/bugs/3862/#4bdd

---

- **labels**: radcan, trigonometry --> radcan, documentation
